### ISSUE TITLE
What do we want to upgrade?

### ISSUE THREAD

**Issue by dirn** [PyGotham/pygotham#113]
The following at the out of date libraries used by PyGotham. I'd like to do any upgrades prior to launching this year's site.

```
alembic (Current: 0.6.5 Latest: 0.7.5.post2 [sdist])
arrow (Current: 0.4.4 Latest: 0.5.4 [sdist])
bleach (Current: 1.4 Latest: 1.4.1 [sdist])
cached-property (Current: 0.1.5 Latest: 1.1.0 [wheel])
decorator (Current: 3.4.0 Latest: 3.4.2 [sdist])
Flask-Admin (Current: 1.0.8 Latest: 1.1.0 [sdist])
Flask-Mail (Current: 0.9.0 Latest: 0.9.1 [sdist])
Flask-Migrate (Current: 1.2.0 Latest: 1.3.1 [sdist])
Flask-Security (Current: 1.7.3 Latest: 1.7.4 [sdist])
Flask-SQLAlchemy (Current: 1.0 Latest: 2.0 [sdist])
Flask-WTF (Current: 0.10.0 Latest: 0.11 [wheel])
jsmin (Current: 2.0.11 Latest: 2.1.1 [sdist])
Mako (Current: 1.0.0 Latest: 1.0.1 [sdist])
psycopg2 (Current: 2.5.3 Latest: 2.6 [sdist])
python-dateutil (Current: 2.2 Latest: 2.4.2 [wheel])
python-slugify (Current: 0.0.7 Latest: 1.0.2 [sdist])
SQLAlchemy (Current: 0.9.7 Latest: 1.0.0 [sdist])
SQLAlchemy-Utils (Current: 0.26.7 Latest: 0.30.0 [sdist])
toolz (Current: 0.7.0 Latest: 0.7.1 [sdist])
Unidecode (Current: 0.4.16 Latest: 0.4.17 [sdist])
validators (Current: 0.6.0 Latest: 0.7 [sdist])
Werkzeug (Current: 0.9.6 Latest: 0.10.4 [wheel])
WTForms (Current: 2.0.1 Latest: 2.0.2 [wheel])
WTForms-Alchemy (Current: 0.12.8 Latest: 0.13.1 [sdist])
WTForms-Components (Current: 0.9.4 Latest: 0.9.7 [sdist])
```

**Comment by dirn** [comment]
The only one that makes me apprehensive is SQLAlchemy 1. I ran into some issues with it at work and had to adjust some logic. I don't foresee any problems here, though.

My personal recommendation is to officially upgrade everything exception SQLAlchemy and for everyone to do all their work to ready the 2015 site using the upgrade SQLAlchemy locally.

We need tests :cry:

**Comment by jonafato** [comment]
TL;DR: All but cached-property and SQLAlchemy can be upgraded right now.

- alembic: Looks fine, I know that 0.7.4 is quite stable, and the branch DAG feature (introduced in [0.7.0](https://alembic.readthedocs.org/en/latest/changelog.html#change-0.7.0-feature)) is definitely useful to have.

- arrow: changes in that range are [mostly bug fixes](https://github.com/crsmithdev/arrow/blob/master/HISTORY.md). +1 to upgrading.

- bleach: patch release, [Python 3.4](https://github.com/jsocol/bleach/blob/master/CHANGES#L15). Upgrade.

- cached-property: breaking changes at [1.0.0](https://github.com/pydanny/cached-property/blob/master/HISTORY.rst). Should be upgraded, adds new TTL feature, but needs further inspection.

- decorator: few actual [changes](https://github.com/micheles/decorator/blob/a4a31e3e6f4149a3e4c071b6a80d593a8658af52/CHANGES.txt#L4-10), upgrade.

- Flask-Admin: lots of bugfixes, some additional features ([changelog](https://flask-admin.readthedocs.org/en/v1.1.0/changelog/)), upgrade.

- Flask-Mail: [bug fixes](https://pythonhosted.org/Flask-Mail/changelog.html), upgrade.

- Flask-Migrate: mostly stylistic and testing changes, new alembic feature support ([changelog](https://github.com/miguelgrinberg/Flask-Migrate/commits/master)), upgrade.

- Flask-Security: [bug fixes](https://github.com/mattupstate/flask-security/blob/develop/CHANGES), upgrade.

- Flask-SQLAlchemy: seemingly [no breaking changes](http://flask-sqlalchemy.pocoo.org/2.0/changelog/) upgrade, but possibly be prepared to rollback or fix.

- Flask-WTF: mostly [bug fixes](https://flask-wtf.readthedocs.org/en/latest/changelog.html), upgrade.

- jsmin: [bug fixes](https://bitbucket.org/dcs/jsmin/src/e0fa31661e18aaaa37285f22ae708dde3880a8d8/CHANGELOG.txt?at=default), upgrade.

- Mako: [minor changes](http://docs.makotemplates.org/en/latest/changelog.html#change-1.0.1), used by Alembic, upgrade.

- psycopg2: [jsonb support](https://github.com/psycopg/psycopg2/blob/master/NEWS), upgrade and trash the whole schema in favor of using jsonb as a document store.

- python-dateutil: [zoneinfo updates, bug fixes, cleanup](https://github.com/dateutil/dateutil/blob/master/NEWS), update.

- python-slugify: [production-ready](https://github.com/un33k/python-slugify/blob/master/CHANGELOG.md), upgrade.

- SQLAlchemy: Per [the recent regressions](https://twitter.com/sqlalchemy/status/590296932481765376), we should probably wait for 1.0.1.

- SQLAlchemy-Utils: [useful features](https://github.com/kvesteri/sqlalchemy-utils/blob/master/CHANGES.rst), no breaking changes, upgrade.

- toolz: patch release, not covered in the [changelog](https://github.com/pytoolz/toolz/blob/master/release-notes), probably can't hurt.

- Unidecode: probably can't hurt.

- validators: [bug fixes](https://github.com/kvesteri/validators/blob/master/CHANGES.rst), upgrade.

- Werkzeug: [many fixes, nothing breaking](http://werkzeug.pocoo.org/docs/0.10/changes/), upgrade.

- WTForms: [localization](https://github.com/wtforms/wtforms/blob/master/CHANGES.rst), upgrade.

- WTForms-Alchemy: [small collection](https://github.com/kvesteri/wtforms-alchemy/blob/master/CHANGES.rst) of useful things, upgrade.

- WTForms-Components: [bug fixes](https://github.com/kvesteri/wtforms-components/blob/master/CHANGES.rst), upgrade.
